decorator.rb

```diff
@@ -0,0 +1,30 @@
+require './nameable'
+
+# Decorator class
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+# Capitalize Decorator
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.upcase
+  end
```

Comment:
- [ ] in this method you are using the `upcase()` method however this will make the entire string uppercase, according to the requirements the method should capitalize the string which is done using the `capitalize()` method

Reply:
@mikethreels Thank you!